com.ibm.streamsx.topology/opt/python/codegen/py_pyTupleTosplTuple.cgt

```diff
@@ -50,10 +50,16 @@ void MY_OPERATOR::fromPythonToPort<%=$oport->getIndex()%>(PyObject *pyTuple <%=$
 <%
     if (defined $iport) {
     
-    # Only copy attributes across if they match on name and type
+    # Only copy attributes across if they match on name and type,
+    # or on name and input type T and output type optional<T>
     my $matchInputAttr = $iport->getAttributeByName($name);
     if (defined $matchInputAttr) {
-       if ($matchInputAttr->getSPLType() eq $attribute->getSPLType()) {
+        my $inputType = $matchInputAttr->getSPLType();
```

Comment:
This code is being duplicated in PR #1562 so once merged that will need to be updated.

Ideally maybe a single perl method.